src/raftstore/store/store.rs

```diff
@@ -433,9 +442,27 @@ impl<T: Transport, C: PdClient> Store<T, C> {
 
     fn on_raft_base_tick(&mut self, event_loop: &mut EventLoop<Self>) {
         let t = Instant::now();
+        let ct = clocktime::now_monotonic_raw_clocktime();
         for (&region_id, peer) in &mut self.region_peers {
             if !peer.get_store().is_applying() {
-                peer.raft_group.tick();
+                if self.cfg.calibrate_tick_with_clocktime {
+                    // Calculate the real tick number with clocktime.
+                    // Since the system load or process/thread scheduling may impact
+                    // the event system, the ticks on eventloop callbacks could be slower than
+                    // it should be. It will calculate the precise tick number when calibrating
+                    // the ticks using the system's monotonic raw clocktime.
+                    let tick_number = calculate_tick_with_clocktime(self.tick_time,
+                                                                    ct,
+                                                                    self.cfg
+                                                                        .raft_base_tick_interval,
+                                                                    peer.is_leader());
+                    // And run `tick()` as many as `tick_number` indicates.
+                    for _ in 0..tick_number {
+                        peer.raft_group.tick();
```

Comment:
Yes, it should be safe to do that. My statement isn't correct. What I want to say actually is this will hurt performance. A slow follower can make cluster perform badly.